### PR TITLE
Get and set multiple registers on Xen

### LINKED ISSUE #133: Get and set mulitple registers in one call
The current implementation of the get and set register functions work only with a single register at a time, which is quite wasteful when multiple registers are needed to be either get or set. If the hypervisor supports accessing all vCPU registers in one shot (like the HVM context on Xen) adding this feature is going to be better performance wise.

### PR DESCRIPTION
**Pull request by tklengyel**
Fixes #133

### CODE REVIEW

File: libvmi/libvmi.h [hunk 1]
```diff
@@ -1856,8 +1900,23 @@ unsigned int vmi_get_num_vcpus (
...
-    reg_t *value,
-    registers_t reg,
+    uint64_t *value,
+    reg_t reg,
+    unsigned long vcpu);
+
+/**
+ * Gets the current value of a VCPU registers.  This currently only
+ * supports x86 registers.  When LibVMI is accessing a raw
+ * memory file or KVM, this function will fail.
```
Comment (smaresca) [inline libvmi/libvmi.h]: Curious: what is needed for arm?

Also, s/Gets the current value of a VCPU registers/Gets the current value of all VCPU registers/

**Comment by tklengyel** [comment]
Retest this please.

File: libvmi/libvmi.h (hunk 1 above)
Comment (tklengyel) [inline libvmi/libvmi.h]: Typo fixed. For ARM we would need an arm_regs struct defined for registers similar to the x86 regs we use here.

**Comment by tklengyel** [comment]
Retest this please.

**Comment by tklengyel** [comment]
Retest this please.

**Comment by tklengyel** [comment]
Retest this please.

**Pull request merged**